Configure should return an error with not supported versions of QT libraries

While configuring with a non-supported version of QT libraries, configure should exit with an error.

Environment:
```
$ git log --oneline -1
9e7dc68 Merge #15616: rpc: Clarify decodescript RPCResult doc
$ ls /usr/lib/x86_64-linux-gnu/libQt5Gui.so.5.3.2
/usr/lib/x86_64-linux-gnu/libQt5Gui.so.5.3.2
$ /usr/lib/x86_64-linux-gnu/qt5/bin/moc --version
moc 5.3.2
$ uname -srv
Linux 4.9.0-0.bpo.7-amd64 #1 SMP Debian 4.9.110-3+deb9u2~deb8u1 (2018-08-14)
```

Steps to replicate:
```
$ ./autogen.sh
$ ./configure --with-incompatible-bdb >/dev/null 2>/dev/null && echo OK
OK
```
Expected behavior: configure should exit with an error due to version of QT libraries not supported.

Details:
```
$ ./configure --with-incompatible-bdb | grep -i qt
configure: WARNING: Found Berkeley DB other than 4.8; wallets opened by this build will not be portable!
checking for QT5... yes
checking for QT_TEST... yes
checking for QT_DBUS... yes
checking for static Qt... no
checking whether -fPIE can be used with this Qt config... no
checking for moc-qt5... no
checking for moc... /usr/lib/x86_64-linux-gnu/qt5/bin/moc
checking for uic-qt5... no
checking for uic... /usr/lib/x86_64-linux-gnu/qt5/bin/uic
checking for rcc-qt5... no
checking for rcc... /usr/lib/x86_64-linux-gnu/qt5/bin/rcc
checking for lrelease-qt5... no
checking for lrelease... /usr/lib/x86_64-linux-gnu/qt5/bin/lrelease
checking for lupdate-qt5... no
checking for lupdate... /usr/lib/x86_64-linux-gnu/qt5/bin/lupdate
checking whether to build Bitcoin Core GUI... yes (Qt5)
configure: WARNING: libzmq version 4.x or greater not found, disabling
checking whether to build test_bitcoin-qt... yes
config.status: creating share/qt/Info.plist
  with gui / qt = yes
```

Yes @MarcoFalke , I've seen the check, but somehow that part of the code is never used (it is used only when there is no pkg-config, and this is forbidden on linux systems). On top of that, in the code there is no evaluation of the QT libraries version when pkg-config is active (again, pkg-config is always active on linux).

I tested it with two systems (Debian Jessie and Debian Stretch), with depends and without, with pkgconfig and without (by embedding in the code not to use pkgconfig, like we do with `$host==*mingw*`) and adding `--with-gui=qt5` or not. Here are the results:

| Environment | Pkgconfig | Depends (5.9.7) | --with-gui= | Result |
| -- | -- | -- | -- | -- |
| Jessie (5.3.2) | Yes | Yes |  | Yes (1) |
| Jessie (5.3.2) | Yes | Yes | qt5 | Yes (3) |
| Jessie (5.3.2) | Yes | No |  | Yes (1) |
| Jessie (5.3.2) | Yes | No | qt5 | Yes (1) |
| Jessie (5.3.2) | No | Yes |  | No (2) |
| Jessie (5.3.2) | No | Yes | qt5 | Error (4) |
| Jessie (5.3.2) | No | No |  | No (2) |
| Jessie (5.3.2) | No | No | qt5 | Error (4) |
| Stretch (5.7.1) | Yes | Yes |  | Yes (1) |
| Stretch (5.7.1) | Yes | Yes | qt5 | Yes (3) |
| Stretch (5.7.1) | Yes | No |  | Yes (1) |
| Stretch (5.7.1) | Yes | No | qt5 | Yes (1) |
| Stretch (5.7.1) | No | Yes |  | No (2) |
| Stretch (5.7.1) | No | Yes | qt5 | Error (4) |
| Stretch (5.7.1) | No | No |  | No (2) |
| Stretch (5.7.1) | No | No | qt5 | Error (4) |

Results:

- Yes (1):
```
checking for QT5... yes
checking for QT_TEST... yes
checking for QT_DBUS... yes
checking for static Qt... no
checking whether -fPIE can be used with this Qt config... no
checking for moc-qt5... no
checking for moc... /usr/lib/x86_64-linux-gnu/qt5/bin/moc
checking for uic-qt5... no
checking for uic... /usr/lib/x86_64-linux-gnu/qt5/bin/uic
checking for rcc-qt5... no
checking for rcc... /usr/lib/x86_64-linux-gnu/qt5/bin/rcc
checking for lrelease-qt5... no
checking for lrelease... /usr/lib/x86_64-linux-gnu/qt5/bin/lrelease
checking for lupdate-qt5... no
checking for lupdate... /usr/lib/x86_64-linux-gnu/qt5/bin/lupdate
checking whether to build Bitcoin Core GUI... yes (Qt5)
checking whether to build test_bitcoin-qt... yes
config.status: creating share/qt/Info.plist
with gui / qt = yes
```
- No (2):
This happens every time there is pkg-config=no (hardcoded by me) and there is no option `--with-gui=qt5`.
```
checking QtPlugin usability... no
configure: WARNING: QtCore headers missing; bitcoin-qt frontend will not be built
checking QtPlugin presence... no
checking for QtPlugin... no
checking whether to build Bitcoin Core GUI... no (Qt5)
with gui / qt = no
```
- Yes (3):
This happens every time pkg-config is activated (it's always pre-activated on linux), and depends is on and there is the option `--with-gui=qt5`.
```
checking for QT5... yes
checking for QT_TEST... yes
checking for QT_DBUS... yes
checking for static Qt... yes
checking for QTFONTDATABASE... yes
checking for QTEVENTDISPATCHER... yes
checking for QTTHEME... yes
checking for QTDEVICEDISCOVERY... yes
checking for QTACCESSIBILITY... yes
checking for QTFB... yes
checking for QTXCBQPA... yes
checking for static Qt plugins: -lqminimal... yes
checking for static Qt plugins: -lqxcb -lxcb-static... yes
checking whether -fPIE can be used with this Qt config... yes
checking for moc-qt5... no
checking for uic-qt5... no
checking for rcc-qt5... no
checking for lrelease-qt5... no
checking for lupdate-qt5... no
checking whether to build Bitcoin Core GUI... yes (Qt5)
checking whether to build test_bitcoin-qt... yes
with gui / qt = yes
```

- Error (4):
This happens every time I use the option `--with-gui=qt5` with pkg-config active (it is always active for linux systems):
```
checking QtPlugin usability... no
checking QtPlugin presence... no
checking for QtPlugin... no
configure: error: QtCore headers missing
```

--- a/build-aux/m4/bitcoin_qt.m4
+++ b/build-aux/m4/bitcoin_qt.m4
@@ -80,10 +80,15 @@ AC_DEFUN([BITCOIN_QT_INIT],[
 ])
 
 dnl Find Qt libraries and includes.
+dnl
+dnl   BITCOIN_QT_CONFIGURE([MINIMUM-VERSION])
+dnl
 dnl Outputs: See _BITCOIN_QT_FIND_LIBS
 dnl Outputs: Sets variables for all qt-related tools.
 dnl Outputs: bitcoin_enable_qt, bitcoin_enable_qt_dbus, bitcoin_enable_qt_test
 AC_DEFUN([BITCOIN_QT_CONFIGURE],[
+  qt_version=">= $1"
+  qt_lib_prefix="Qt5"
   BITCOIN_QT_CHECK([_BITCOIN_QT_FIND_LIBS])
 
   dnl This is ugly and complicated. Yuck. Works as follows:
@@ -221,7 +226,7 @@ AC_DEFUN([BITCOIN_QT_CONFIGURE],[
     bitcoin_enable_qt=no
   ])
   if test x$bitcoin_enable_qt = xyes; then
-    AC_MSG_RESULT([$bitcoin_enable_qt ($QT_LIB_PREFIX)])
+    AC_MSG_RESULT([$bitcoin_enable_qt ($qt_lib_prefix)])
   else
     AC_MSG_RESULT([$bitcoin_enable_qt])
   fi
@@ -315,23 +320,29 @@ dnl Internal. Find Qt libraries using pkg-config.
 dnl Outputs: All necessary QT_* variables are set.
 dnl Outputs: have_qt_test and have_qt_dbus are set (if applicable) to yes|no.
 AC_DEFUN([_BITCOIN_QT_FIND_LIBS],[
-  m4_ifdef([PKG_CHECK_MODULES],[
-    QT_LIB_PREFIX=Qt5
-    qt5_modules="Qt5Core Qt5Gui Qt5Network Qt5Widgets"
-    BITCOIN_QT_CHECK([
-      PKG_CHECK_MODULES([QT5], [$qt5_modules], [QT_INCLUDES="$QT5_CFLAGS"; QT_LIBS="$QT5_LIBS" have_qt=yes],[have_qt=no])
-
-      if test "x$have_qt" != xyes; then
-        have_qt=no
-        BITCOIN_QT_FAIL([Qt dependencies not found])
-      fi
-    ])
-    BITCOIN_QT_CHECK([
-      PKG_CHECK_MODULES([QT_TEST], [${QT_LIB_PREFIX}Test], [QT_TEST_INCLUDES="$QT_TEST_CFLAGS"; have_qt_test=yes], [have_qt_test=no])
-      if test "x$use_dbus" != xno; then
-        PKG_CHECK_MODULES([QT_DBUS], [${QT_LIB_PREFIX}DBus], [QT_DBUS_INCLUDES="$QT_DBUS_CFLAGS"; have_qt_dbus=yes], [have_qt_dbus=no])
-      fi
-    ])
+  BITCOIN_QT_CHECK([
+    PKG_CHECK_MODULES([QT_CORE], [${qt_lib_prefix}Core $qt_version], [],
+                      [BITCOIN_QT_FAIL([${qt_lib_prefix}Core $qt_version not found])])
   ])
-  true; dnl
+  BITCOIN_QT_CHECK([
+    PKG_CHECK_MODULES([QT_GUI], [${qt_lib_prefix}Gui $qt_version], [],
+                      [BITCOIN_QT_FAIL([${qt_lib_prefix}Gui $qt_version not found])])
+  ])
+  BITCOIN_QT_CHECK([
+    PKG_CHECK_MODULES([QT_WIDGETS], [${qt_lib_prefix}Widgets $qt_version], [],
+                      [BITCOIN_QT_FAIL([${qt_lib_prefix}Widgets $qt_version not found])])
+  ])
+  BITCOIN_QT_CHECK([
+    PKG_CHECK_MODULES([QT_NETWORK], [${qt_lib_prefix}Network $qt_version], [],
+                      [BITCOIN_QT_FAIL([${qt_lib_prefix}Network $qt_version not found])])
+  ])
+  QT_INCLUDES="$QT_CORE_CFLAGS $QT_GUI_CFLAGS $QT_WIDGETS_CFLAGS $QT_NETWORK_CFLAGS"
+  QT_LIBS="$QT_CORE_LIBS $QT_GUI_LIBS $QT_WIDGETS_LIBS $QT_NETWORK_LIBS"
+
+  BITCOIN_QT_CHECK([
+    PKG_CHECK_MODULES([QT_TEST], [${qt_lib_prefix}Test $qt_version], [QT_TEST_INCLUDES="$QT_TEST_CFLAGS"; have_qt_test=yes], [have_qt_test=no])
+    if test "x$use_dbus" != xno; then
+      PKG_CHECK_MODULES([QT_DBUS], [${qt_lib_prefix}DBus $qt_version], [QT_DBUS_INCLUDES="$QT_DBUS_CFLAGS"; have_qt_dbus=yes], [have_qt_dbus=no])
+    fi
+  ])
 ])

--- a/build-aux/m4/bitcoin_qt.m4
+++ b/build-aux/m4/bitcoin_qt.m4
@@ -295,25 +295,19 @@ AC_DEFUN([_BITCOIN_QT_FIND_STATIC_PLUGINS],[
       if test -d "$qt_plugin_path/platforms/android"; then
         QT_LIBS="$QT_LIBS -L$qt_plugin_path/platforms/android -lqtfreetype -lEGL"
       fi
-      m4_ifdef([PKG_CHECK_MODULES],[
-        if test x$bitcoin_cv_qt58 = xno; then
-          PKG_CHECK_MODULES([QTPLATFORM], [Qt5PlatformSupport], [QT_LIBS="$QTPLATFORM_LIBS $QT_LIBS"])
-        else
-          PKG_CHECK_MODULES([QTFONTDATABASE], [Qt5FontDatabaseSupport], [QT_LIBS="-lQt5FontDatabaseSupport $QT_LIBS"])
-          PKG_CHECK_MODULES([QTEVENTDISPATCHER], [Qt5EventDispatcherSupport], [QT_LIBS="-lQt5EventDispatcherSupport $QT_LIBS"])
-          PKG_CHECK_MODULES([QTTHEME], [Qt5ThemeSupport], [QT_LIBS="-lQt5ThemeSupport $QT_LIBS"])
-          PKG_CHECK_MODULES([QTDEVICEDISCOVERY], [Qt5DeviceDiscoverySupport], [QT_LIBS="-lQt5DeviceDiscoverySupport $QT_LIBS"])
-          PKG_CHECK_MODULES([QTACCESSIBILITY], [Qt5AccessibilitySupport], [QT_LIBS="-lQt5AccessibilitySupport $QT_LIBS"])
-          PKG_CHECK_MODULES([QTFB], [Qt5FbSupport], [QT_LIBS="-lQt5FbSupport $QT_LIBS"])
-        fi
-        if test "x$TARGET_OS" = xlinux; then
-          PKG_CHECK_MODULES([QTXCBQPA], [Qt5XcbQpa], [QT_LIBS="$QTXCBQPA_LIBS $QT_LIBS"])
-        elif test "x$TARGET_OS" = xdarwin; then
-          PKG_CHECK_MODULES([QTCLIPBOARD], [Qt5ClipboardSupport], [QT_LIBS="-lQt5ClipboardSupport $QT_LIBS"])
-          PKG_CHECK_MODULES([QTGRAPHICS], [Qt5GraphicsSupport], [QT_LIBS="-lQt5GraphicsSupport $QT_LIBS"])
-          PKG_CHECK_MODULES([QTCGL], [Qt5CglSupport], [QT_LIBS="-lQt5CglSupport $QT_LIBS"])
-        fi
-      ])
+      PKG_CHECK_MODULES([QTFONTDATABASE], [Qt5FontDatabaseSupport], [QT_LIBS="-lQt5FontDatabaseSupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTEVENTDISPATCHER], [Qt5EventDispatcherSupport], [QT_LIBS="-lQt5EventDispatcherSupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTTHEME], [Qt5ThemeSupport], [QT_LIBS="-lQt5ThemeSupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTDEVICEDISCOVERY], [Qt5DeviceDiscoverySupport], [QT_LIBS="-lQt5DeviceDiscoverySupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTACCESSIBILITY], [Qt5AccessibilitySupport], [QT_LIBS="-lQt5AccessibilitySupport $QT_LIBS"])
+      PKG_CHECK_MODULES([QTFB], [Qt5FbSupport], [QT_LIBS="-lQt5FbSupport $QT_LIBS"])
+      if test "x$TARGET_OS" = xlinux; then
+        PKG_CHECK_MODULES([QTXCBQPA], [Qt5XcbQpa], [QT_LIBS="$QTXCBQPA_LIBS $QT_LIBS"])
+      elif test "x$TARGET_OS" = xdarwin; then
+        PKG_CHECK_MODULES([QTCLIPBOARD], [Qt5ClipboardSupport], [QT_LIBS="-lQt5ClipboardSupport $QT_LIBS"])
+        PKG_CHECK_MODULES([QTGRAPHICS], [Qt5GraphicsSupport], [QT_LIBS="-lQt5GraphicsSupport $QT_LIBS"])
+        PKG_CHECK_MODULES([QTCGL], [Qt5CglSupport], [QT_LIBS="-lQt5CglSupport $QT_LIBS"])
+      fi
     fi
 ])
 

--- a/configure.ac
+++ b/configure.ac
@@ -1134,7 +1134,7 @@ else
   BITCOIN_QT_INIT
 
   dnl sets $bitcoin_enable_qt, $bitcoin_enable_qt_test, $bitcoin_enable_qt_dbus
-  BITCOIN_QT_CONFIGURE
+  BITCOIN_QT_CONFIGURE([5.5.1])
 fi
 
 if test x$enable_wallet != xno; then

